ComponentKit/Core/CKComponentLifecycleManager.h

```diff
@@ -60,6 +60,13 @@ extern const CKComponentLifecycleManagerState CKComponentLifecycleManagerStateEm
 - (void)updateWithState:(const CKComponentLifecycleManagerState &)state;
 
 /**
+ Updates the context to the new one. 
+ 
+ Component provider will receive new context value next time the component tree will be built.
+ */
+- (void)updateContext:(id)newContext;
```

Comment:
I would not add mutable state to the CLM.

We might as well take out the context from the lifecycle manager, and pass it as an extra argument of prepareForUpdateWithModel. The context should be treated as any props at this layer.